packages/twenty-server/src/engine/api/graphql/workspace-query-runner/factories/query-result-getters.factory.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable } from '@nestjs/common';
+
+import { ObjectMetadataInterface } from 'src/engine-metadata/field-metadata/interfaces/object-metadata.interface';
+
+import { TokenService } from 'src/engine/modules/auth/services/token.service';
+
+@Injectable()
+export class QueryResultGettersFactory {
+  constructor(private readonly tokenService: TokenService) {}
+
+  async create<Result>(
+    result: Result,
+    objectMetadataItem: ObjectMetadataInterface,
+  ): Promise<Result> {
+    // TODO: look for file type once implemented
+    switch (objectMetadataItem.nameSingular) {
+      case 'attachment':
+        return this.applyAttachmentGetters(result);
+      default:
+        return result;
+    }
+  }
+
+  private async applyAttachmentGetters<Result>(
+    attachments: any,
+  ): Promise<Result> {
+    if (!attachments || !attachments.edges) {
+      return attachments;
+    }
+
+    const mappedEdges = await Promise.all(
+      attachments.edges.map(async (attachment: any) => {
+        if (attachment?.node?.fullPath) {
+          const todayDate = new Date();
+          const expirationDate = todayDate.setDate(todayDate.getDate() + 1);
```

Comment:
to discuss how much time we want the url valid

Reply:
1 day is fine for me. From a product perspective what would be nice would be to display an informative message in that scenario that indicates the link has expired and not just a random white screen/redirect.
Maybe best to have it as an env var since it's what we've done so far? Or at least someplace where we group constants even if they can't be overwritten by env?

Reply:
@FelixMalfait maybe a dedicated route?

Reply:
Thank you both, changes have been applied